File: Source/Binary.FSharp.CSharp/GeneratorBuilderFSharpExtensions.cs
using Mikodev.Binary.Collections;
using Mikodev.Binary.Unions;
using System;

namespace Mikodev.Binary
{
    public static class GeneratorBuilderFSharpExtensions
    {
        public static IGeneratorBuilder AddFSharpConverterCreators(this IGeneratorBuilder builder)
        {
            if (builder is null)
                throw new ArgumentNullException(nameof(builder));
            _ = builder.AddConverterCreator(new FSharpCollectionConverterCreator());
            _ = builder.AddConverterCreator(new UnionConverterCreator());
            return builder;
        }
    }
}

File: Source/Binary.FSharp/GeneratorBuilderFSharpExtensions.fs
namespace Mikodev.Binary

open Mikodev.Binary.Internal.Creators
open Mikodev.Binary.Internal.Creators.Collections
open System.Runtime.CompilerServices

[<Extension>]
type GeneratorBuilderFSharpExtensions =
    [<Extension>]
    static member AddFSharpConverterCreators(builder : IGeneratorBuilder) =
        builder
            .AddConverterCreator(UnionConverterCreator())
            .AddConverterCreator(ListConverterCreator())
            .AddConverterCreator(MapConverterCreator())
            .AddConverterCreator(SetConverterCreator())
